File: tests/fixtures/swift/sample.swift
import Foundation
import XCTest

/// A base protocol.
public protocol Repo: Sendable {
    func find(id: Int) -> User?
    var count: Int { get }
}

/// A user model.
public struct User: Codable, Equatable {
    let id: Int
    var name: String
    static let empty = User(id: 0, name: "")
}

enum Status: String {
    case active, inactive
    case pending = "p"
}

open class BaseRepo {
    fileprivate var cache: [Int: User] = [:]
    public init() {}
    func load() {}
}

/// User repository.
final class UserRepo: BaseRepo, Repo {
    private let session: Session
    var count: Int { return cache.count }
    override init() { self.session = Session(); super.init() }
    public func find(id: Int) -> User? {
        let u = User(id: id, name: "x")
        let token = ProcessInfo.processInfo.environment["API_TOKEN"]
        session.get(url: "x", retries: 2)
        return self.cache[id] ?? helper(u)
    }
    private static func helper(_ u: User) async throws -> User { return u }
    @objc func fooObjc() {}
}

extension UserRepo {
    func extra() -> Int { return load2() }
}

typealias UserMap = [Int: User]
let GLOBAL_CONST = 3
var mutableGlobal = "a"

actor Counter {
    var value = 0
    func inc() { value += 1 }
}

func topLevel(a: Int, b: UserRepo) -> String { return "\(a)" }

class UserRepoTests: XCTestCase {
    func testFind() { let r = UserRepo(); XCTAssertNotNil(r.find(id: 1)) }
}

File: tests/fixtures/dart/sample.dart
import 'package:flutter/material.dart';
import 'package:myapp/models/user.dart' as models show User, Session hide Foo;
import 'utils.dart';
import '../core/base.dart';
export 'src/widgets.dart';
part 'sample.g.dart';

/// Base repository.
abstract class BaseRepo<T> implements Disposable {
  void load();
}

mixin Loggable {
  void log(String m) {}
}

/// User repository.
class UserRepo extends BaseRepo<User> with Loggable implements Repo, Other {
  final Session session;
  String? _name;
  static const int MAX = 3;
  List<User> users = [];

  UserRepo(this.session);
  UserRepo.named(this.session) : super();
  factory UserRepo.create() => UserRepo(Session());

  int get count => users.length;
  set name(String v) => _name = v;

  @override
  void load() {
    final u = User(id: 1);
    var s = Session.open('x');
    session.get('url', retries: 2);
    _helper(u);
    models.User.fromJson({});
    final token = Platform.environment['API_TOKEN'];
    print(token);
  }

  Future<User?> find(int id) async => users.first;
  void _helper(User u) {}
}

enum Status { active, inactive, pending }

extension UserExt on User {
  String greet() => 'hi';
}

typedef Callback = void Function(int);
typedef OldCb(int x);

const int GLOBAL_MAX = 5;
final config = Config();
var counter = 0;

Future<void> topLevel(UserRepo repo, {int retries = 1}) async {
  await repo.find(1);
  topLevel2();
}

void main() {
  group('UserRepo', () {
    test('finds user', () {
      final r = UserRepo(Session());
      expect(r.find(1), isNotNull);
    });
  });
}
